sim/midas/src/main/cc/bridges/cpu_managed_stream.h

```diff
@@ -39,14 +43,15 @@ typedef struct CPUManagedStreamParameters {
  * FPGA-managed AXI4 for their platform.
  *
  */
-class CPUManagedStream {
+class CPUManagedDriver {
 public:
-  CPUManagedStream(CPUManagedStreamParameters params,
+  CPUManagedDriver(StreamParameters params,
                    std::function<uint32_t(size_t)> mmio_read_func)
```

Comment:
Instead of a function, could we pass a pointer to the `simif_emul_t` interface or whatever base class provides `read` and `write` methods? In case one does not exist, maybe one should be created. I think in this case a lambda is a bit too generic.

Reply:
The thing is this is used across both FPGA and metasim. I'll give it some more thought. If provide a type signature for a lambda, fundamentally i should be able to define a common interface and mix it into both classes.